install/kubernetes/istio-auth-with-cluster-ca.yaml

```diff
@@ -207,17 +207,33 @@ spec:
         - mountPath: /etc/certs
           name: istio-certs
           readOnly: true
+        - mountPath: /etc/istio/ingress-certs
+          name: ingress-certs
+          readOnly: true
         - name: config-volume
           mountPath: /etc/istio/config
       volumes:
       - name: istio-certs
         secret:
           secretName: istio.default
+      - name: ingress-certs
+        secret:
+          secretName: istio-ingress-certs
       - name: config-volume
         configMap:
           name: istio
 ---
-
+################################
+# Empty secret for Istio ingress controller
+################################
+apiVersion: v1
+kind: Secret
+metadata:
+  name: istio-ingress-certs
+data:
+  tls.key:
+  tls.crt:
+---
```

Comment:
Why is this needed?

Reply:
Is it to prevent the container from not starting until the CA generates the secret?

Reply:
You are partially right - this is to prevent container from not starting.
The ingress controller TLS key/cert (for external traffic) is not managed by CA. The operator needs to replace the secret with the real ones when TLS is enabled for external traffic.
To be specific, the operators needs to do:
kubectl delete secret istio-ingress-certs; kubectl create secret tls istio-ingress-certs --key /tmp/tls.key --cert /tmp/tls.crt